tests/factory/test_factory_unused_options.py

```diff
@@ -0,0 +1,127 @@
+import copy
+import os
+
+import pytest
+
+from mozdownload.utils import urljoin
+from mozdownload import FactoryScraper
+
+
+@pytest.mark.parametrize('filename,args', [
+    # ReleaseScraper
+    ('release',
+     {
+         'fname': 'firefox-23.0.1.en-US.win32.exe',
+         'kwargs': {
+             'version': '23.0.1',
+             'platform': 'win32',
+             # unused options
+             'branch': 'mozilla-central',
+             'build_id': '20131001030204',
+             'build_number': '1',
+             'revision': '8fcac92cfcad',
+             'date': '2013-10-01',
+             'debug_build': True,
+             'url': 'http://localhost',
+         }
+     }),
+
+    # ReleaseCandidateScraper
+    ('candidate',
+     {
+         'fname': 'firefox-23.0.1-build3.en-US.win32.exe',
+         'kwargs': {
+             'platform': 'win32',
+             'version': '23.0.1',
+             # unused options
+             'branch': 'mozilla-central',
+             'build_id': '20131001030204',
+             'revision': '8fcac92cfcad',
+             'date': '2013-10-01',
+             'debug_build': True,
+             'url': 'http://localhost',
+         }
+     }),
+
+    # DailyScraper
+    ('daily',
+     {
+         'fname': '2013-10-01-03-02-04-mozilla-central-firefox-27.0a1.en-US.win32.installer.exe',
+         'kwargs': {
+             'platform': 'win32',
+             # unused options
+             'debug_build': True,
+             'url': 'http://localhost',
+             'version': '23.0.1',
+         }
+     }),
+
+    # TinderboxScraper
+    ('tinderbox',
+     {
+         'fname': '1374583608-mozilla-central-firefox-25.0a1.en-US.win32.installer.exe',
+         'kwargs': {
+             'platform': 'win32',
+             # unused options
+             'url': 'http://localhost',
+             'version': '23.0.1',
+         }
+     }),
+
+    # TryScraper
+    ('try',
+     {
+         'builds': ['/firefox/try-builds/test-user@mozilla.com-8fcac92cfcad/try-foobar/'],
+         'fname': '8fcac92cfcad-firefox-38.0a1.en-US.mac.dmg',
+         'kwargs': {
+             'revision': '8fcac92cfcad',
+             'platform': 'mac64',
+             # unused options
+             'build_id': '20131001030204',
+             'build_number': '1',
+             'url': 'http://localhost',
+             'version': '23.0.1',
+         }
+     })
+])
+def test_unknown_options(httpd, tmpdir, filename, args, mocker):
```

Comment:
The filename of this test file is `unused`, while here you have `unknown`. Please fix those method names.

Reply:
incorporated this change in the latest commit. Please Review.